pyproject.toml

```diff
@@ -1,7 +1,98 @@
 [build-system]
-requires = ["jupyter_packaging~=0.7.0", "jupyterlab>=3.0.0,==3.*", "setuptools>=40.8.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = [
+    "hatchling",
+    "jupyterlab>=3.0.0,==3.*",
+]
+build-backend = "hatchling.build"
+
+[project]
+name = "ipympl"
+description = "Matplotlib Jupyter Extension"
+readme = "README.md"
+license = {file = "LICENSE"}
+authors = [
+    {name = "Matplotlib Development Team", email = "matplotlib-users@python.org"},
+]
+keywords = [
+    "graphics",
+    "ipython",
+    "jupyter",
+    "widgets",
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Framework :: IPython",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Programming Language :: Python :: 3.5",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
```

Comment:
Should we add Python 3.12 here?

Reply:
Yes, I don't know why I stopped at 3.11.

We should probably decide on what Python versions we are supporting, heading toward [SPEC0](https://scientific-python.org/specs/spec-0000/) eventually. Maybe 3.9 to 3.12 is appropriate now. Then I could ensure that CI is running using both the earliest and latest Python versions that we claim to support.

Reply:
Sounds good 👍🏽